### ISSUE TITLE
BaseListView: Specifying sorter attribute labels

### ISSUE THREAD

**Issue by vercotux** [yiisoft/yii2#9828]
The BaseListView widget has a [$sorter](http://www.yiiframework.com/doc-2.0/yii-widgets-baselistview.html#$sorter-detail) attribute which takes in an array to configure the sorter widget.
By default the sorter widget used by BaseListView is [yii\widgets\LinkSorter](http://www.yiiframework.com/doc-2.0/yii-widgets-linksorter.html).

LinkSorter in turn provides an [$attributes](http://www.yiiframework.com/doc-2.0/yii-widgets-linksorter.html#$attributes-detail) property, which can only take in a simple list of attributes and nothing more, so we can use something like the following when initializing ListView:

```
'sorter'            => [
    'attributes'    => [
        'id',
        'created_at',
        'user_id',
    ],
],
```

According to the documentation, when $attributes in LinkSorter is left blank, a different initialization is used:

> If not set, it will be determined using [yii\data\Sort::$attributes](http://www.yiiframework.com/doc-2.0/yii-data-sort.html#$attributes-detail).

One would assume that it would be possible to (for example) set a label directly from ListView by setting the 'sorter' property in the following fashion:

```
'sorter'            => [
    'attributes'    => [
        'created_at'    => [
            'asc'       => ['created_at' => SORT_ASC],
            'desc'      => ['created_at' => SORT_DESC],
            'default'   => SORT_ASC,
            'label'     => 'Time',
        ],
    ],
],
```

The above however yields the following error in [getAttributeOrder() of Sort.php](https://github.com/yiisoft/yii2/blob/95ba06c01b35b84740cd566edf4bdc0611db1efa/framework/data/Sort.php#L279):

> Illegal offset type in isset or empty

**Comment by jeicd** [comment]
Well, the current implementation of LinkSorter accepts only a one dimensional ```attributes``` array property and it is only possible to reorder attributes or manage their visibility.

**Comment by vercotux** [comment]
Indeed, but at the very least it should support setting labels, perhaps in the form of
`[ attribute => label ]`

What is currently the best practice to override a label from a view file?

**Comment by njitram** [comment]
+1

**Comment by samdark** [comment]
Won't be implemented in 2.0.
